Sources/Mini/Store.swift

```diff
@@ -1,138 +1,30 @@
-/*
- Copyright [2019] [BQ]
-
- Licensed under the Apache License, Version 2.0 (the "License");
- you may not use this file except in compliance with the License.
- You may obtain a copy of the License at
-
- http://www.apache.org/licenses/LICENSE-2.0
-
- Unless required by applicable law or agreed to in writing, software
- distributed under the License is distributed on an "AS IS" BASIS,
- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- See the License for the specific language governing permissions and
- limitations under the License.
- */
-
+import Combine
 import Foundation
-import RxSwift
-
-public protocol StoreType {
-    associatedtype State: StateType
-    associatedtype StoreController: Disposable
 
-    var state: State { get set }
-    var dispatcher: Dispatcher { get }
-    var reducerGroup: ReducerGroup { get }
-
-    func replayOnce()
-}
-
-extension StoreType {
-    /**
-     Property responsible of reduce the `State` given a certain `Action` being triggered.
-     ```
-     public var reducerGroup: ReducerGroup {
-        ReducerGroup {[
-            Reducer(of: SomeAction.self, on: self.dispatcher) { (action: SomeAction)
-                self.state = myCoolNewState
-            },
-            Reducer(of: OtherAction.self, on: self.dispatcher) { (action: OtherAction)
-                // Needed work
-                self.state = myAnotherState
-                }
-            }
-        ]}
-     ```
-     - Note : The property has a default implementation which complies with the @_functionBuilder's current limitations, where no empty blocks can be produced in this iteration.
-     */
-    public var reducerGroup: ReducerGroup {
-        return ReducerGroup()
-    }
-}
-
-public class Store<State: StateType, StoreController: Disposable>: ObservableType, StoreType {
-    public typealias Element = State
+@available(iOS 13.0, *)
+public class Store<State: ObservableObject, StoreController: Cancellable>: ObservableObject {
```

Comment:
`State` must conform to `StateType`, if it has to conform to `ObservableObject`, delegate it to the former protocol.

Reply:
I do not understand the need for the StateType and its implication in these classes. Right now in our flow it seems irrelevant. I would appreciate if you could tell me a bit more about this and see if I understand why to include it here.

Reply:
Because working with protocols allow extensions to be done into generics while classes don't. It's a matter of extensibility.

Reply:
Ok, protocol restored and applied at both State and Store.